dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,79 +234,124 @@ std::wstring Microsoft::Windows::AppNotifications::Helpers::SetDisplayNameBasedO
     return displayName;
 }
 
-void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring const& appId, std::wstring const& clsid)
+// Placeholder
+HRESULT RetrieveAssetsFromProcess(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
 {
-    wil::unique_hkey hKey;
-    // subKey: \Software\Classes\AppUserModelId\{AppGUID}
-    std::wstring subKey{ c_appIdentifierPath + appId };
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
 
-    THROW_IF_WIN32_ERROR(RegCreateKeyEx(
-        HKEY_CURRENT_USER,
-        subKey.c_str(),
-        0,
-        nullptr /* lpClass */,
-        REG_OPTION_NON_VOLATILE,
-        KEY_ALL_ACCESS,
-        nullptr /* lpSecurityAttributes */,
-        &hKey,
-        nullptr /* lpdwDisposition */));
+    return E_NOTIMPL;
+}
 
-    // Retrieve the display name
-    std::wstring displayName{};
-    displayName = SetDisplayNameBasedOnProcessName();
+// Do nothing. This is just a placeholder while the UDK is ingested with the proper API.
+HRESULT ToastNotifications_RetrieveAssets_Stub(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
+{
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
+
+    return E_NOTIMPL;
+}
+
+HRESULT RetrieveAssetsFromWindow(_Out_ PWSTR* displayName, _Out_ PWSTR* iconFilePath)
+{
+    *displayName = nullptr;
+    *iconFilePath = nullptr;
 
-    // Retrieve the icon
-    std::wstring iconFilePath{};
+    wil::unique_cotaskmem_string localDisplayName;
+    wil::unique_cotaskmem_string localIconFilePath;
 
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+    HWND hWindow = GetForegroundWindow();
```

Comment:
Will this work if app window is minimized?

Reply:
This doesn't seem right. What are you doing with this information?  The foreground window is any window in the system that has foreground, not just the one for this app.  Did you mean GetActiveWindow? Even then that might not be right.

Reply:
Can you have the caller pass you context somehow?  Coming soon we'll have a `WindowId` construct that the caller can provide that is a reference to an HWND you can then use the HWNDy functions on directly.

Reply:
@jonwis Given that WinUI apps are not running under a console window, we still need to get its window. What would be the right API to use here?

Also, when the `WindowId` functionality will be available for use?

Reply:
You should probably talk with @rkarman or @jeffstall with a comprehensive "what are you really trying to do" question.  

My main issue here is the _implicit_ context being used. Instead, consider an API where the app passes you the asset/text information necessary.

Reply:
After discussing with the team, for now I'm going to remove the window check and re-introduce it at a later release.

Reply:
Agree with this decision. We should focus on the Process Icon, Shortcut icon and default icon. Everything else is a lower priority IMHO.